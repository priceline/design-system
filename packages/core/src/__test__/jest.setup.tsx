import '@testing-library/jest-dom'
import 'jest-styled-components'

global.requestAnimationFrame =
  global.requestAnimationFrame ||
  function _raf(cb) {
    return setTimeout(cb, 0)
  }
