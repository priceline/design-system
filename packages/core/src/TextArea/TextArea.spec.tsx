import { render } from '../__test__/testing-library'
import { TextArea } from '..'

const id = 'fake-test-id'

describe('TextArea', () => {
  test('it renders', () => {
    const { asFragment } = render(<TextArea id={id} />)
    expect(asFragment()).toMatchSnapshot()
  })
  test('it renders an input element with a red border with a color prop is passed', () => {
    const { asFragment } = render(<TextArea id={id} color='error' />)
    expect(asFragment()).toMatchSnapshot()
  })
  test('it renders an input element with a really large padding and margin', () => {
    const { asFragment } = render(<TextArea id={id} p={4} m={4} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
