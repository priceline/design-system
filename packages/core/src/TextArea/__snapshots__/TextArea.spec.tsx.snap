// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea it renders 1`] = `
<DocumentFragment>
  .c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c1 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  width: 100%;
  font-family: inherit;
  font-size: 14px;
  color: inherit;
  background-color: transparent;
  border-radius: 12px;
  border-width: 1px;
  border-style: solid;
  padding-top: 14px;
  padding-bottom: 14px;
  padding-left: 12px;
  padding-right: 12px;
  margin: 0;
  border-color: #c0cad5;
}

.c1::-webkit-input-placeholder {
  color: #4f6f8f;
}

.c1::-moz-placeholder {
  color: #4f6f8f;
}

.c1:-ms-input-placeholder {
  color: #4f6f8f;
}

.c1::placeholder {
  color: #4f6f8f;
}

.c1::-ms-clear {
  display: none;
}

.c1:focus {
  outline: 0;
  border-color: #0068ef;
  box-shadow: 0 0 0 2px #0068ef;
}

<div
    class="c0"
  >
    <textarea
      class="c1"
      id="fake-test-id"
    />
  </div>
</DocumentFragment>
`;

exports[`TextArea it renders an input element with a really large padding and margin 1`] = `
<DocumentFragment>
  .c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c1 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  width: 100%;
  font-family: inherit;
  font-size: 14px;
  color: inherit;
  background-color: transparent;
  border-radius: 12px;
  border-width: 1px;
  border-style: solid;
  padding-top: 14px;
  padding-bottom: 14px;
  padding-left: 12px;
  padding-right: 12px;
  margin: 0;
  border-color: #c0cad5;
  padding: 32px;
  margin: 32px;
}

.c1::-webkit-input-placeholder {
  color: #4f6f8f;
}

.c1::-moz-placeholder {
  color: #4f6f8f;
}

.c1:-ms-input-placeholder {
  color: #4f6f8f;
}

.c1::placeholder {
  color: #4f6f8f;
}

.c1::-ms-clear {
  display: none;
}

.c1:focus {
  outline: 0;
  border-color: #0068ef;
  box-shadow: 0 0 0 2px #0068ef;
}

<div
    class="c0"
  >
    <textarea
      class="c1"
      id="fake-test-id"
    />
  </div>
</DocumentFragment>
`;

exports[`TextArea it renders an input element with a red border with a color prop is passed 1`] = `
<DocumentFragment>
  .c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c1 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  width: 100%;
  font-family: inherit;
  font-size: 14px;
  color: inherit;
  background-color: transparent;
  border-radius: 12px;
  border-width: 1px;
  border-style: solid;
  padding-top: 14px;
  padding-bottom: 14px;
  padding-left: 12px;
  padding-right: 12px;
  margin: 0;
  border-color: #c00;
}

.c1::-webkit-input-placeholder {
  color: #4f6f8f;
}

.c1::-moz-placeholder {
  color: #4f6f8f;
}

.c1:-ms-input-placeholder {
  color: #4f6f8f;
}

.c1::placeholder {
  color: #4f6f8f;
}

.c1::-ms-clear {
  display: none;
}

.c1:focus {
  outline: 0;
  border-color: #c00;
  box-shadow: 0 0 0 2px #c00;
}

<div
    class="c0"
  >
    <textarea
      class="c1"
      color="error"
      id="fake-test-id"
    />
  </div>
</DocumentFragment>
`;
